.github/workflows/main_build.yml

```diff
@@ -0,0 +1,37 @@
+name: CryptoAnalysis build
+
+on: [push, pull_request]
```

Comment:
maybe restrict this to develop and main only. Feature braches don't need to waste runner performence.

Reply:
I added it like this so that forks also run the tests on their branches.

Reply:
We can try it like that first. If we run into issues, we will restrict this.